consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/CliqueHelpers.java

```diff
@@ -90,4 +92,36 @@ private static int minimumBlocksSincePreviousSigning(
     final int signerLimit = (validatorCount / 2) + 1;
     return signerLimit - 1;
   }
+
+  public static void installCliqueBlockChoiceRule(
+      final Blockchain blockchain, final CliqueContext cliqueContext) {
+    blockchain.setBlockChoiceRule(
+        blockchain
+            // most total difficulty
+            .getBlockChoiceRule()
+            // shortest chain
+            .thenComparing(Comparator.comparing(ProcessableBlockHeader::getNumber).reversed())
+            // proposer with least recent in-turn block
+            .thenComparing((BlockHeader header) -> distanceFromInTurn(header, cliqueContext))
+            // last resort: blockhash as uint256.
+            .thenComparing(Comparator.comparing(BlockHeader::getHash).reversed()));
+  }
+
+  static int distanceFromInTurn(final BlockHeader header, final CliqueContext context) {
```

Comment:
reading the spec I'm still confused - maybe it's worth putting the formula in for context? 

Clients should choose the block with the largest value for this calculation:

(header_number - validator_index) % validator_count

Reply:
added javadoc.